lib/bootloader.js

```diff
@@ -45,7 +45,12 @@
   // version without applicationCache intercepting us.
   function getShadowIndex(success, failure) {
     var req = new XMLHttpRequest();
-    req.open('GET', 'APPSHELL_PATHappshell.html', true);
+    var path = 'APPSHELL_PATHappshell';
+    var query = window.location.search.match(/revision=([^&#]+)/);
```

Comment:
Either way this can be handled in separate PR, either it's needed or it isn't and has no effect.